pkg/tfbridge/property_path.go

```diff
@@ -2,11 +2,37 @@ package tfbridge
 
 import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
 	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/walk"
 )
 
+// a variant of PropertyPath.Get which works on PropertyMaps
+func getPathFromPropertyMap(
+	path resource.PropertyPath, propertyMap resource.PropertyMap,
+) (resource.PropertyValue, bool) {
+	if len(path) == 0 {
+		return resource.NewNullProperty(), false
+	}
+
+	rootKeyStr, ok := path[0].(string)
+	contract.Assertf(ok && rootKeyStr != "", "root key must be a non-empty string")
+	rootKey := resource.PropertyKey(rootKeyStr)
+	restPath := path[1:]
+
+	if len(restPath) == 0 {
+		return propertyMap[rootKey], true
+	}
+
+	if !propertyMap.HasValue(rootKey) {
+		return resource.NewNullProperty(), false
+	}
+
+	return restPath.Get(propertyMap[rootKey])
+}
```

Comment:
I would remove this. If you need it, the "variant" nature makes me very concerned. The behavior of `getPathFromPropertyMap` isn't consistent with `resource.PropertyPath.Get`:
1. `resource.PropertyPath{}.Get(v)` returns `v, true`. `getPathFromPropertyMap` returns a null property.
2. `path[0]` must be a non-empty string. If not, the function panics. Why doesn't it just return false. `""` is a valid key for a property map (maps represent objects and actual maps), so panicing feels like a bad idea here.

```suggestion
// a variant of PropertyPath.Get which works on PropertyMaps
func getPathFromPropertyMap(
	path resource.PropertyPath, propertyMap resource.PropertyMap,
) (resource.PropertyValue, bool) {
	 return path.Get(resource.NewProperty(propertyMap))
}
```

In general, variants need to explain how they vary from the function they resemble, and why that's necessary.